examples/server.rs

```diff
@@ -4,58 +4,224 @@
+/// ./example-server -i red
+///
+/// To specify an alternate pixel format, use the `--big-endian` flag and/or the ordering flags. The
+/// server will transform the input image/color to the requested pixel format and use the format
+/// for the RFB protocol.
+///
+/// For example, to use big-endian xRGB:
+/// ./example-server --big-endian true -r 1 -g 2 -b 3
+///
+struct Args {
+    /// Image/color to display from the server
+    #[clap(value_enum, short, long, default_value_t = Image::Oxide)]
+    image: Image,
+
+    /// Pixel endianness
+    #[clap(long, default_value_t = false, action = clap::ArgAction::Set)]
+    big_endian: bool,
+
+    /// Byte mapping to red (4-byte RGB pixel, endian-agnostic)
+    #[clap(short, long, default_value_t = 0)]
+    red_order: u8,
+
+    /// Byte mapping to green (4-byte RGB pixel, endian-agnostic)
+    #[clap(short, long, default_value_t = 1)]
+    green_order: u8,
+
+    /// Byte mapping to blue (4-byte RGB pixel, endian-agnostic)
+    #[clap(short, long, default_value_t = 2)]
+    blue_order: u8,
+}
+
+#[derive(ValueEnum, Debug, Copy, Clone)]
+enum Image {
+    Oxide,
+    TestTubes,
+    Red,
+    Green,
+    Blue,
+    White,
+    Black,
+}
+
+#[derive(Clone)]
+struct ExampleServer {
+    display: Image,
+    rgb_order: (u8, u8, u8),
+    big_endian: bool,
+}
+
 #[tokio::main]
-async fn main() {
+async fn main() -> Result<()> {
     env_logger::init();
 
+    let args = Args::parse();
+    validate_order(args.red_order, args.green_order, args.blue_order)?;
+
+    let pf = PixelFormat::new_colorformat(
+        rgb_888::BITS_PER_PIXEL,
+        rgb_888::DEPTH,
+        args.big_endian,
+        order_to_shift(args.red_order),
+        rgb_888::MAX_VALUE,
+        order_to_shift(args.green_order),
+        rgb_888::MAX_VALUE,
+        order_to_shift(args.blue_order),
+        rgb_888::MAX_VALUE,
+    );
+    info!(
+        "Starting server: image: {:?}, pixel format; {:#?}",
+        args.image, pf
+    );
+
     let config = VncServerConfig {
         addr: SocketAddr::new(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)), 9000),
         version: ProtoVersion::Rfb38,
         sec_types: SecurityTypes(vec![SecurityType::None, SecurityType::VncAuthentication]),
         name: "rfb-example-server".to_string(),
     };
     let data = VncServerData {
-        width: 1024,
-        height: 768,
+        width: WIDTH as u16,
+        height: HEIGHT as u16,
+        input_pixel_format: pf.clone(),
+    };
+    let server = ExampleServer {
+        display: args.image,
+        rgb_order: (args.red_order, args.green_order, args.blue_order),
+        big_endian: args.big_endian,
     };
-    let server = ExampleServer {};
     let s = VncServer::new(server, config, data);
     s.start().await;
+
+    Ok(())
 }
 
-#[derive(Clone)]
-struct ExampleServer {}
+fn validate_order(r: u8, g: u8, b: u8) -> Result<()> {
+    if r > 3 || g > 3 || b > 3 {
+        return Err(anyhow!("r/g/b must have ordering of 0, 1, 2, or 3"));
```

Comment:
anyhow also has a `bail!` macro so you don't have to prefix `return Err(` a bunch

Reply:
neat! thanks for that suggestion.

done

Reply:
yea, np. though, i should've been more clear haha. You don't need to nest them, using only `bail!(...)` suffices

Reply:
ah, got it, thanks.

done